### PR TITLE
rgw: add support for caching of Keystone admin token.

### PR DESCRIPTION
**Pull request by rzarzynski**

### CODE REVIEW

**Comment by rzarzynski** [comment]
@yehudasa This PR has been verified using Tempest, s3-tests and manually. I'm removing the _DNM_ tag right now.

**Comment by yehudasa** [comment]
@rzarzynski this PR will need to be refactored on top of the v3 code

**Comment by rzarzynski** [comment]
@yehudasa rework is in progress.

**Comment by rzarzynski** [comment]
@yehudasa I've rebased and retested this PR after merging the Keystone v3 support.

File: src/rgw/rgw_keystone.h [hunk 1]
```diff
@@ -6,6 +6,12 @@
 
 #include "rgw_common.h"
 
+int open_cms_envelope(CephContext *cct, string& src, string& dst);
```
Comment (yehudasa) [inline src/rgw/rgw_keystone.h]: @rzarzynski please add `rgw_` prefix to all the utility functions that are not static anymore

Comment (rzarzynski) [inline src/rgw/rgw_keystone.h]: @yehudasa added.

**Comment by smithfarm** [comment]
Changelog:
* created http://tracker.ceph.com/issues/16110 to track the new feature
* added tracker URL to PR description

**Pull request merged**